### PR TITLE
add config for gke soak e2e job

### PR DESCRIPTION
**Pull request by j3ffml**
Forgot to check this change in, test has been running config from my fork.

### CODE REVIEW

**Comment by ixdy** [comment]
What creates this soak cluster? Is it done manually out-of-band as needed?

**Comment by jlowdermilk** [comment]
http://kubekins.dls.corp.google.com/view/GKE/job/kubernetes-soak-weekly-deploy-gke/ creates it

**Comment by ixdy** [comment]
Can we add kubernetes-soak-weekly-deploy-gke to e2e.sh as well, then?

**Comment by jlowdermilk** [comment]
Deploy job config added.

**Comment by jlowdermilk** [comment]
@k8s-bot test this

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -691,6 +691,36 @@ case ${JOB_NAME} in
           )"}
     ;;
 
+  # Sets up the GCE soak cluster weekly using the latest CI release.
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: nit: GKE

Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: Done.

**Comment by ixdy** [comment]
LGTM

**Comment by jlowdermilk** [comment]
Change after LGTM was bumping node size to n1-standard-2 so perf tests can run. Re-adding.

**Pull request merged**